core/data/src/main/java/com/nolawiworkineh/data/auth/EncryptedSessionStorage.kt

```diff
@@ -1,49 +1,54 @@
-package com.nolawiworkineh.core.data.auth
+package com.nolawiworkineh.data.auth
 
 import android.content.SharedPreferences
-import com.nolawiworkineh.core.domain.AuthInfo.AuthInfo
 import com.nolawiworkineh.core.domain.SessionStorage
+import com.nolawiworkineh.core.domain.authInfo.AuthInfo
 import kotlinx.coroutines.Dispatchers
 import kotlinx.coroutines.withContext
 import kotlinx.serialization.encodeToString
 import kotlinx.serialization.json.Json
+import timber.log.Timber
 
 
 class EncryptedSessionStorage(
     private val sharedPreferences: SharedPreferences,
-): SessionStorage {
+) : SessionStorage {
 
-    //  Retrieves and decrypts the stored AuthInfo if available.
     override suspend fun get(): AuthInfo? {
         return withContext(Dispatchers.IO) {
-            val json = sharedPreferences.getString(KEY_AUTH_INFO, null)
-            json?.let {
-                // Deserialize the JSON back into AuthInfoSerializable, then map it to AuthInfo
-                Json.decodeFromString<AuthInfoSerializable>(it).toAuthInfo()
+            try {
+                val json = sharedPreferences.getString(KEY_AUTH_INFO, null)
+                json?.let {
+                    Json.decodeFromString<AuthInfoSerializable>(it).toAuthInfo()
+                }
+            } catch (e: Exception) {
+                Timber.e(e, "Failed to retrieve auth info")
+                null
             }
```

Comment:
Be careful with catching general exceptions inside suspend functions. You can break coroutine cancellation that way. If a coroutine is cancelled, the currently suspending function (here, it would be `withContext()`) will throw a CancellationException. If your catch block eats that up, this exception won't be properly propagated up the coroutine scopes which can cause some coroutines to run longer than they should.
You can fix this by adding this:

```
if(e is CancellationException) throw e
```
or alternatively by only catching specific exceptions and not CancellationExceptions.

In this video I show the issue with that a bit more in detail (starting at 17:45):
https://www.youtube.com/watch?v=VWlwkqmTLHc